### PR TITLE
11.0 mig account_cutoff_accrual_base

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please squash in all your PRs together adjacent "OCA Transbot..." commits.

File: account_cutoff_accrual_base/README.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License
+
+Account Accrual Base
+====================
+
+This module contains objets, fields and menu entries that are used by other
```
Comment (simahawk) [inline account_cutoff_accrual_base/README.rst]: `s/objets/models`

File: account_cutoff_accrual_base/README.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+accrual modules. So you need to install other accrual modules to get the
+additionnal functionalities:
+
+* the module 'account_cutoff_accrual_picking' will manage accrued expenses and
+  revenues based on pickings.
+* a not-developped-yet module will manage accrued expenses and revenues based
+  on timesheets.
+
+Please contact Alexis de Lattre from Akretion <alexis.delattre@akretion.com>
```
Comment (simahawk) [inline account_cutoff_accrual_base/README.rst:18]: update README w/ latest OCA version https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: account_cutoff_accrual_base/__init__.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+##############################################################################
```
Comment (simahawk) [inline account_cutoff_accrual_base/__init__.py]: remove this header

File: account_cutoff_accrual_base/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
+##############################################################################
```
Comment (simahawk) [inline account_cutoff_accrual_base/__manifest__.py]: use shorter header

File: account_cutoff_accrual_base/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+
+{
+    'name': 'Account Accrual Base',
+    'version': '11.0.1.0.0',
+    'category': 'Accounting & Finance',
+    'license': 'AGPL-3',
+    'summary': 'Base module for accrued expenses and revenues',
+    'author': "Akretion,Odoo Community Association (OCA)",
+    'website': 'http://www.akretion.com',
```
Comment (simahawk) [inline account_cutoff_accrual_base/__manifest__.py]: https://github.com/OCA/account-closing

File: account_cutoff_accrual_base/models/__init__.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+##############################################################################
```
Comment (simahawk) [inline account_cutoff_accrual_base/models/__init__.py]: same

File: account_cutoff_accrual_base/models/account.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
+##############################################################################
```
Comment (simahawk) [inline account_cutoff_accrual_base/models/account.py]: same

File: account_cutoff_accrual_base/models/account_cutoff.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
+##############################################################################
```
Comment (simahawk) [inline account_cutoff_accrual_base/models/account_cutoff.py]: same for all the py files from now on :)

File: account_cutoff_accrual_base/models/account_cutoff.py [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+
+from openerp import api, fields, models
```
Comment (simahawk) [inline account_cutoff_accrual_base/models/account_cutoff.py]: `s/openerp/odoo` for all files

File: account_cutoff_accrual_base/models/account_cutoff.py [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+import openerp.addons.decimal_precision as dp
+
+
+class AccountCutOff(models.Model):
+    _inherit = 'account.cutoff'
+
+    @api.model
+    def _inherit_default_cutoff_account_id(self):
+        account_id = super()._inherit_default_cutoff_account_id()
+        type = self.env.context.get('type')
```
Comment (simahawk) [inline account_cutoff_accrual_base/models/account_cutoff.py]: `type_` or `cutoff_type`

File: account_cutoff_accrual_base/tests/__init__.py [hunk 11]
```diff
@@ -0,0 +1,5 @@
+##############################################################################
```
Comment (simahawk) [inline account_cutoff_accrual_base/tests/__init__.py]: remove

File: account_cutoff_accrual_base/tests/test_account_cutoff.py [hunk 12]
```diff
@@ -0,0 +1,28 @@
+##############################################################################
```
Comment (simahawk) [inline account_cutoff_accrual_base/tests/test_account_cutoff.py]: replace w/ short header

File: account_cutoff_accrual_base/tests/test_account_cutoff.py [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    def test_inherit_default_cutoff_account_id(self):
+        company = self.env.user.company_id
+        default_accrued_expense_account_id = \
+            company.default_accrued_expense_account_id.id or False
+        default_accrued_revenue_account_id = \
+            company.default_accrued_revenue_account_id.id or False
+
+        account_id = \
+            self.env['account.cutoff'].with_context(type='accrued_expense').\
+            _inherit_default_cutoff_account_id()
```
Comment (simahawk) [inline account_cutoff_accrual_base/tests/test_account_cutoff.py]: `_inherit_default_cutoff_account_id` should be removed. See review on cutoff base module #61

**Review by simahawk — CHANGES_REQUESTED** [review]
you must add dependency to cutoff base PR to fix tests

File: account_cutoff_accrual_base/models/account.py [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import fields, models
```
Comment (mpanarin) [inline account_cutoff_accrual_base/models/account.py]: change to odoo?

**Review by mpanarin — CHANGES_REQUESTED** [review]

**Comment by fclementic2c** [comment]
:+1:

**Pull request merged**